### PR TITLE
[TypeDeclaration] Skip ArrayDimFetch for return and param types

### PR DESCRIPTION
**Pull request by staabm**
as proposed in https://github.com/rectorphp/rector-src/pull/5056#issuecomment-1729360011

### CODE REVIEW

**Comment by staabm** [comment]
@samsonasik please see the failing tests

**Comment by samsonasik** [comment]
should be covered by https://github.com/rectorphp/rector-src/pull/5058, please rebase

File: rules-tests/TypeDeclaration/Rector/ClassMethod/AddMethodCallBasedStrictParamTypeRector/Fixture/skip_phpdoc_shape.php.inc [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+final class SkipPhpdocArrayShape {
+    private function doFoo() {
+        $shape = $this->returnsShape();
+        $this->doBar($shape['hello']);
+    }
+
+    /**
+     * @return array{'hello': string, 'world': int}
+     */
+    private function returnsShape() {
```
Comment (samsonasik) [inline rules-tests/TypeDeclaration/Rector/ClassMethod/AddMethodCallBasedStrictParamTypeRector/Fixture/skip_phpdoc_shape.php.inc:14]: for note, if you have native return `: array` type defined combined with docblock, then it is what it is :), as you got what you defined

**Comment by samsonasik** [comment]
It seems green, I am merging it ;)

File: rules-tests/TypeDeclaration/Rector/ClassMethod/AddMethodCallBasedStrictParamTypeRector/Fixture/skip_phpdoc_shape.php.inc (hunk 1 above)
Comment (samsonasik) [reply rules-tests/TypeDeclaration/Rector/ClassMethod/AddMethodCallBasedStrictParamTypeRector/Fixture/skip_phpdoc_shape.php.inc:14]: oh, as it detect by native type of ArrayDimFetch var, it seems already covered :)

**Comment by staabm** [comment]
Excellent - thanks 🙏

**Pull request merged**